### ISSUE TITLE
Incorrect guessit

### ISSUE THREAD

**Issue by OmgImAlexis** [guessit-io/guessit#398]
```
File /tv/Beyond/Beyond - S01E07 - The Hour of the Wolf 720p WEB-DL.mkv is same size as /downloads/Beyond.S01E02.Tempus.Fugit.720p.FREE.WEBRip.AAC2.0.x264-BTW/gNWDXow11s7E0X7GTDrZ.mkv
```

```
For: /downloads/Beyond.S01E02.Tempus.Fugit.720p.FREE.WEBRip.AAC2.0.x264-BTW/gNWDXow11s7E0X7GTDrZ.mkv
GuessIt found: {
    "title": "Beyond", 
    "season": 1, 
    "episode_title": "Tempus Fugit", 
    "screen_size": "720p", 
    "format": "WEBRip", 
    "audio_codec": "AAC", 
    "audio_channels": "2.0", 
    "video_codec": "h264", 
    "release_group": "BTW", 
    "episode": 7, 
    "container": "mkv", 
    "mimetype": "video/x-matroska", 
    "type": "episode"
}
```

**Comment by OmgImAlexis** [comment]
```
For: /downloads/We.Bare.Bears.S01E14.Brother.Up.1080p.WEB-DL.AAC2.0.H.264-TVSmash/mxNMuJWeO7PUWCMEwqKSsS6D8Vs9S6V3PHD.mkv 
GuessIt found: {
    "title": "We Bare Bears", 
    "episode": 14, 
    "episode_title": "Brother Up", 
    "screen_size": "1080p", 
    "format": "WEB-DL", 
    "audio_codec": "AAC", 
    "audio_channels": "2.0", 
    "video_codec": "h264", 
    "release_group": "TVSmash", 
    "season": 6, 
    "container": "MKV", 
    "type": "episode"
}
```

**Comment by duramato** [comment]
My suggestion to fix this is create an hash propriety of sorts, to the similiarity of the crc32 there currently is. 
After a quick google i found this https://github.com/SmeegeSec/HashTag which might aid this "task".
And if guessit would detect an hash, it wouldn't be able to make other matches from it like it is in this issue. Like the Season 6 from the `S6` in `mxNMuJWeO7PUWCMEwqKSsS6D8Vs9S6V3PHD`

**Comment by ratoaq2** [comment]
HashTag is not reliable. It identifies the second scenario as `osCommerce` which is wrong.

My first question is: From where these file names comes from? Is that a standard, convention or something? If so, then I would like to see more examples.

**Comment by duramato** [comment]
The files are from Usenet, uploaders de-identify the files by using hashes, to avoid DMCA/EUCD takedown bots, and you only know which are which by using the indexer that made since they have the I'd for them.The folder name comes from the nzb name they have it indexed for. Now for the hash algorithms they use it's a mystery.. each indexer/or each uploader uses their own.

**Comment by ratoaq2** [comment]
Hash-Algorithm-Identifier might help. It can identify several hash algorithms and it uses regexes.

```
Enter the Hash : gNWDXow11s7E0X7GTDrZ

Most Probable Hash Algorithms found:

[+] Lotus Domino
________________________________________________________________________________


Enter the Hash : mxNMuJWeO7PUWCMEwqKSsS6D8Vs9S6V3PHD

Most Probable Hash Algorithms found:

[+] SSHA-1
```

https://github.com/AnimeshShaw/Hash-Algorithm-Identifier/blob/master/hashidentifier/HashIdentifier.py

Maybe guessit could detect only the common hashes. What do you think @Toilal ?

Are we able to get more examples?

**Comment by ratoaq2** [comment]
Just some brainstorming...

Another way would be:
Discard all matches from filename (except extension)
if filename is only 1 word with season or episode matches in it and folder name contains more matches including season or episode in it.

**Comment by ratoaq2** [comment]
Fixed by #405
